base/sparse/sparsematrix.jl

```diff
@@ -942,8 +943,9 @@ function getindex{Tv,Ti<:Integer}(A::SparseMatrixCSC{Tv,Ti}, I::Range, J::Abstra
     # Form the structure of the result and compute space
     for j = 1:nJ
         @inbounds col = J[j]
+        1 <= col <= n || throw(BoundsError())
         @simd for k in colptrA[col]:colptrA[col+1]-1
-            if rowvalA[k] in I; nnzS += 1 end # `in` is fast for ranges
+            @inbounds nnzS += rowvalA[k] in I # `in` is fast for ranges
```

Comment:
This had `@simd` but not `@inbounds`, so it wasn't vectorizable. I haven't checked to see if LLVM is actually vectorizing it now, but it's conceivably possible.